AWSS3/AWSS3TransferUtility+EnumerateBlocks.m

```diff
@@ -0,0 +1,43 @@
+//
+// Copyright 2010-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+#import "AWSS3TransferUtility+EnumerateBlocks.h"
+
+@implementation AWSS3TransferUtility (EnumerateBlocks)
+
+- (void)enumerateToAssignBlocks:(AWSS3TransferUtilityBlocks *)blocks {
+    [self enumerateToAssignBlocksForUploadTask:^(AWSS3TransferUtilityUploadTask * _Nonnull uploadTask, AWSS3TransferUtilityProgressBlock  _Nullable __autoreleasing * _Nullable uploadProgressBlockReference, AWSS3TransferUtilityUploadCompletionHandlerBlock  _Nullable __autoreleasing * _Nullable completionHandlerReference) {
+        *uploadProgressBlockReference = blocks.uploadProgressBlock;
+        *completionHandlerReference = blocks.uploadCompletedBlock;
+    } multiPartUploadBlocksAssigner:^(AWSS3TransferUtilityMultiPartUploadTask * _Nonnull multiPartUploadTask, AWSS3TransferUtilityMultiPartProgressBlock  _Nullable __autoreleasing * _Nullable multiPartUploadProgressBlockReference, AWSS3TransferUtilityMultiPartUploadCompletionHandlerBlock  _Nullable __autoreleasing * _Nullable completionHandlerReference) {
```

Comment:
Why do we need two `_Nullable` for the reference?
Can we write this same as the function definition?

```swift
[self enumerateToAssignBlocksForUploadTask:^(AWSS3TransferUtilityUploadTask *uploadTask,
                                                 AWSS3TransferUtilityProgressBlock *uploadProgressBlockReference,
                                                 AWSS3TransferUtilityUploadCompletionHandlerBlock  *completionHandlerReference) {
        
        *uploadProgressBlockReference = blocks.uploadProgressBlock;
        *completionHandlerReference = blocks.uploadCompletedBlock;
        
    } multiPartUploadBlocksAssigner:^(
                                      AWSS3TransferUtilityMultiPartUploadTask *multiPartUploadTask,
                                      AWSS3TransferUtilityMultiPartProgressBlock *multiPartUploadProgressBlockReference,
                                      AWSS3TransferUtilityMultiPartUploadCompletionHandlerBlock *completionHandlerReference
                                      ) {
        
        *multiPartUploadProgressBlockReference = blocks.multiPartUploadProgressBlock;
```

Reply:
I got to this code by letting Xcode expand it for me. All the details in this syntax may not be necessary but I did not want to strip it out since this appears to be a sensitive area with the compiler. I will trim out some details since normally these nullable details are only needed with public headers to integrate with Swift. As an internal detail it does not matter to Objective-C.